csi/identity.go

```diff
@@ -0,0 +1,64 @@
+/*
+CSI Interface for OSD
+Copyright 2017 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package csi
+
+import (
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"golang.org/x/net/context"
+)
+
+const (
+	csiDriverVersion = "0.1.0"
+	csiDriverName    = "csi-osd"
+)
+
+var (
+	csiVersion = &csi.Version{
+		Major: 0,
+		Minor: 0,
+		Patch: 0,
+	}
+)
+
+func (s *OsdCsiServer) GetSupportedVersions(
+	context.Context, *csi.GetSupportedVersionsRequest) (*csi.GetSupportedVersionsResponse, error) {
+	return &csi.GetSupportedVersionsResponse{
+		Reply: &csi.GetSupportedVersionsResponse_Result_{
+			Result: &csi.GetSupportedVersionsResponse_Result{
+				SupportedVersions: []*csi.Version{
+					csiVersion,
+				},
+			},
+		},
+	}, nil
+}
+
+func (s *OsdCsiServer) GetPluginInfo(
+	context.Context, *csi.GetPluginInfoRequest) (*csi.GetPluginInfoResponse, error) {
+	return &csi.GetPluginInfoResponse{
+		Reply: &csi.GetPluginInfoResponse_Result_{
+			Result: &csi.GetPluginInfoResponse_Result{
+				Name:          csiDriverName,
+				VendorVersion: csiDriverVersion,
+				Manifest: map[string]string{
+					"hello":                  "world",
```

Comment:
What would be the real data here in this map?

Reply:
Don't know yet. Most likely be `nil` for the moment.